### PR TITLE
(#119) Throws matcher now takes a message matcher

### PR DESCRIPTION
**Pull request by victornoel**
This is for #119.

I went with making message a `Matcher` so that we can match messages in anyway we want, including without any message.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#126](https://github.com/llorllale/cactoos-matchers/issues/126) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#126](https://github.com/llorllale/cactoos-matchers/issues/126) is assigned to @vzurauskas[/z](https://www.0crat.com/u/vzurauskas), here is [why](https://www.0crat.com/footprint/C63314D6Z/85c1eb1e-a525-4910-a99f-9ba4262585e0); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/main/java/org/llorllale/cactoos/matchers/Throws.java [hunk 1]
```diff
@@ -88,33 +114,18 @@ protected boolean matchesSafely(final Scalar<T> obj,
             matches = false;
             dsc.appendText("The exception wasn't thrown.");
         } catch (final Exception cause) {
-            describe(dsc, cause.getClass(), cause.getMessage());
+            dsc
+                .appendText("Exception has type ")
+                .appendValue(cause.getClass())
```
Comment (vzurauskas) [inline src/main/java/org/llorllale/cactoos/matchers/Throws.java]: @victornoel Consider changing this to `cause.getClass().getName()` to keep this part of description behaviour unchanged.

File: src/main/java/org/llorllale/cactoos/matchers/Throws.java [hunk 2]
```diff
@@ -88,33 +114,18 @@ protected boolean matchesSafely(final Scalar<T> obj,
...
+                .appendText(" and message ")
```
Comment (vzurauskas) [inline src/main/java/org/llorllale/cactoos/matchers/Throws.java]: @victornoel Consider using single quotes. Then already existing unit tests won't need to be changed.

File: src/test/java/org/llorllale/cactoos/matchers/ThrowsTest.java [hunk 3]
```diff
@@ -126,8 +126,36 @@ public void describeExpectedValues() {
...
+                    + " and message matches \"NPE\""
+            )
+        ).affirm();
+    }
+
+    @Test
+    public void noMessageMatchesException() {
+        new Assertion<>(
+            // @checkstyle LineLength (1 line)
+            "must match if message is no present",
```
Comment (vzurauskas) [inline src/test/java/org/llorllale/cactoos/matchers/ThrowsTest.java]: @victornoel This line doesn't seem to be long enough to require checkstyle comment.

Comment (vzurauskas) [inline src/test/java/org/llorllale/cactoos/matchers/ThrowsTest.java]: @victornoel "no present" -> "not present".

**Comment by vzurauskas** [comment]
@llorllale This PR also solves https://github.com/llorllale/cactoos-matchers/issues/108. There's [my PR](https://github.com/llorllale/cactoos-matchers/pull/123) for that, but I think the way @victornoel solved it here is better. What do you think?

File: src/main/java/org/llorllale/cactoos/matchers/Throws.java (hunk 1 above)
Comment (victornoel) [reply src/main/java/org/llorllale/cactoos/matchers/Throws.java]: @vzurauskas good point, I changed too much things at once

File: src/main/java/org/llorllale/cactoos/matchers/Throws.java (hunk 2 above)
Comment (victornoel) [reply src/main/java/org/llorllale/cactoos/matchers/Throws.java]: @vzurauskas good point, I changed too much things at once

**Comment by victornoel** [comment]
@vzurauskas thanks, I've updated my PR

**Comment by vzurauskas** [comment]
@llorllale Looks good, see also [this comment](https://github.com/llorllale/cactoos-matchers/pull/126#issuecomment-497378541).

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 29 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vzurauskas[/z](https://www.0crat.com/u/vzurauskas), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#126](https://github.com/llorllale/cactoos-matchers/issues/126) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +25 point(s) just awarded to @vzurauskas[/z](https://www.0crat.com/u/vzurauskas)

**Comment by vzurauskas** [comment]
@sereshqua Does the rule about starting all comments with an `@` reference to a person also apply to all code review comments?

**Comment by victornoel** [comment]
@vzurauskas it does apply for all comments

**Comment by vzurauskas** [comment]
@victornoel Ok, thanks.

**Comment by llorllale** [comment]
@rultor release, tag is `0.16`

**Pull request merged**